From 5ff257a156eda518c1e61d93df3c28e4c430ca56 Mon Sep 17 00:00:00 2001
From: JunSung Choi <junsungc@gooroom.kr>
Date: Sat, 25 May 2019 18:06:33 +0900
Subject: [PATCH 30/63] add 'setTrustMode()' path in GooroomPref

---
 gooroom/browser/gooroom_pref.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 2c240ec3b..5210e3356 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -86,6 +86,13 @@ void GooroomPref::setup() {
     if (command.find("--user-data-dir") !=
         std::string::npos) {  // exist --user-data-dir
       use_user_data_dir = true;
+
+      // bug patch, setting trust mode when the browser launched without any url.
+      if(command == untrust_user_data_dir_option)
+        base::GetGooroomBase()->setTrustMode(false);
+      else
+        base::GetGooroomBase()->setTrustMode(true);
+
       break;
     }
     if (GURL(*it).is_valid() == true) {
-- 
2.11.0

