From 669710ba38b7fcd6d1ad184fe9e21d71b30055c0 Mon Sep 17 00:00:00 2001
From: JunSung Choi <junsungc@gooroom.kr>
Date: Sat, 25 May 2019 21:02:12 +0900
Subject: [PATCH 44/63] Support ppapi-flash module

---
 gooroom/browser/gooroom_pref.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 7370b4590..ac9eef74d 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -77,6 +77,8 @@ void GooroomPref::setup() {
   std::string untrust_user_data_dir_option =
       "--user-data-dir=" + home_path_dir.AsUTF8Unsafe() +
       "/.config/gooroom/untrust";
+  std::string flash_dir = "--ppapi-flash-path=/usr/lib/adobe-flashplugin/libpepflashplayer.so";
+  std::string flash_version = "--ppapi-flash-version=31.0.0.153";
 
   bool use_user_data_dir = false;
 
@@ -106,7 +108,7 @@ void GooroomPref::setup() {
           exit(0);
         } else {
           execl(exe_path.c_str(), "gooroom-browser",
-                trust_user_data_dir_option.c_str(), GURL(*it).spec().c_str(),
+                trust_user_data_dir_option.c_str(), flash_dir.c_str(), flash_version.c_str(), GURL(*it).spec().c_str(),
                 NULL);
           exit(0);
         }
@@ -116,7 +118,7 @@ void GooroomPref::setup() {
           exit(0);
         } else {
           execl(exe_path.c_str(), "gooroom-browser",
-                untrust_user_data_dir_option.c_str(), GURL(*it).spec().c_str(),
+                untrust_user_data_dir_option.c_str(),flash_dir.c_str(), flash_version.c_str(), GURL(*it).spec().c_str(),
                 NULL);
           exit(0);
         }
-- 
2.11.0

