From b7ef2f97cd7ddb3d2bd25ee4850a5689861c9e3b Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Wed, 4 Sep 2019 20:22:29 +0900
Subject: [PATCH 08/78] move GooroomBase and GooroomUrlFilter to base module #2

---
 gooroom/BUILD.gn | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/gooroom/BUILD.gn b/gooroom/BUILD.gn
index 4c01bffacc45..9022675bc760 100644
--- a/gooroom/BUILD.gn
+++ b/gooroom/BUILD.gn
@@ -254,10 +254,6 @@ if (!is_android && !is_mac) {
         "browser/gooroom_browser_extra_parts_security.h",
 
         # for commit #4
-        "base/gooroom_base.cc",
-        "base/gooroom_base.h",
-        "base/gooroom_url_filter.cc",
-        "base/gooroom_url_filter.h",
         "browser/gooroom_pref.cc",
         "browser/gooroom_pref.h",
         "browser/gooroom_url_filter_impl.cc",
-- 
2.20.1

