From 070a5d4b499da2d698b0cfadc066133d5d936575 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Thu, 23 Jul 2020 17:11:59 +0900
Subject: [PATCH 20/84] add 'setTrustMode()' path in GooroomPref

---
 gooroom/browser/gooroom_pref.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index d0de138d5..8d5bd131d 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -90,6 +90,13 @@ void GooroomPref::setup() {
     if (command.find("--user-data-dir") !=
         std::string::npos) {  // exist --user-data-dir
       use_user_data_dir = true;
+
+      // bug patch, setting trust mode when the browser launched without any url.
+      if(command == untrust_user_data_dir_option)
+        base::GetGooroomBase()->setTrustMode(false);
+      else
+        base::GetGooroomBase()->setTrustMode(true);
+
       break;
     }
     if (GURL(*it).is_valid() == true) {
-- 
2.20.1

