From 69867e3dc59993d5a2a13809b035239159baeda8 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Thu, 4 Jun 2020 16:30:33 +0900
Subject: [PATCH 24/79] add 'setTrustMode()' path in GooroomPref

---
 gooroom/browser/gooroom_pref.cc | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 115ac6f499e9..22ed4a438c54 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -88,6 +88,13 @@ void GooroomPref::setup() {
     if (command.find("--user-data-dir") !=
         std::string::npos) {  // exist --user-data-dir
       use_user_data_dir = true;
+
+      // bug patch, setting trust mode when the browser launched without any url.
+      if(command == untrust_user_data_dir_option)
+        base::GetGooroomBase()->setTrustMode(false);
+      else
+        base::GetGooroomBase()->setTrustMode(true);
+
       break;
     }
     if (GURL(*it).is_valid() == true) {
-- 
2.20.1

