From 6102df1dfe971a6310b44c03a18bcfdba60bb362 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Fri, 5 Jun 2020 10:25:13 +0900
Subject: [PATCH 31/79] change default path of policy including content 
 settings

---
 chrome/common/chrome_paths.cc | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/chrome/common/chrome_paths.cc b/chrome/common/chrome_paths.cc
index f72c03f88244..012fc0e3c607 100644
--- a/chrome/common/chrome_paths.cc
+++ b/chrome/common/chrome_paths.cc
@@ -4,6 +4,9 @@
 
 #include "chrome/common/chrome_paths.h"
 
+#if defined(GOOROOM_BUILD)
+#include "gooroom/base/gooroom_base.h"
+#endif
 #include "base/files/file_util.h"
 #include "base/logging.h"
 #include "base/native_library.h"
@@ -498,6 +501,19 @@ bool PathProvider(int key, base::FilePath* result) {
     case chrome::DIR_POLICY_FILES: {
 #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/opt/chrome/policies"));
+#elif defined(GOOROOM_BUILD)
+//change gooroom policies path
+      base::FilePath home_path_dir;
+      base::PathService::Get(base::DIR_HOME, &home_path_dir);
+      std::string assignedDir =
+          home_path_dir.AsUTF8Unsafe() + "/.config/gooroom/trust";
+
+      if (base::GetGooroomBase()->isTrustMode())
+        cur = base::FilePath(
+            FILE_PATH_LITERAL("/usr/share/gooroom/browser/policies/trust"));
+      else
+        cur = base::FilePath(
+            FILE_PATH_LITERAL("/usr/share/gooroom/browser/policies/untrust"));
 #else
       cur = base::FilePath(FILE_PATH_LITERAL("/etc/chromium/policies"));
 #endif
-- 
2.20.1

