From b07303b5e91d074880af6dd71e61d17468d9ed2f Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Thu, 5 Sep 2019 15:24:29 +0900
Subject: [PATCH 37/78] add 'setTrustMode()' path in GooroomPref

---
 gooroom/browser/gooroom_pref.cc | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index fc630b18b96f..0004fcf6008f 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -79,6 +79,19 @@ void GooroomPref::setup() {
       "/.config/gooroom/untrust";
 
   bool use_user_data_dir = false;
+
+  base::FilePath user_data_dir = cmd->GetSwitchValuePath("user-data-dir");
+  if (!user_data_dir.empty()) {
+    use_user_data_dir = true;
+    if (user_data_dir.AsUTF8Unsafe() ==
+        home_path_dir.AsUTF8Unsafe() + "/.config/gooroom/untrust")
+      base::GetGooroomBase()->setTrustMode(false);
+    else if (user_data_dir.AsUTF8Unsafe() !=
+             home_path_dir.AsUTF8Unsafe() + "/.config/gooroom/trust")
+      base::GetGooroomBase()->setTrustMode(true);
+    return;
+  }
+
   for (base::CommandLine::StringVector::const_iterator it = argv.begin();
        it != argv.end(); ++it) {
     //DLOG(ERROR) << *it;
-- 
2.20.1

