From 8b4aadfbf4dbdbbb3d8b532c05f993b71f6f9764 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc91@gmail.com>
Date: Fri, 23 Apr 2021 10:16:19 +0900
Subject: [PATCH 53/63] Change syslog id

---
 base/syslog_logging.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/syslog_logging.cc b/base/syslog_logging.cc
index a043727ae8743..ba2c05a9cdd73 100644
--- a/base/syslog_logging.cc
+++ b/base/syslog_logging.cc
@@ -148,7 +148,7 @@ EventLogMessage::~EventLogMessage() {
   if (user_sid != nullptr)
     ::LocalFree(user_sid);
 #elif BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
-  const char kEventSource[] = "chrome";
+  const char kEventSource[] = "gooroom-browser";
   openlog(kEventSource, LOG_NOWAIT | LOG_PID, LOG_USER);
   // We can't use the defined names for the logging severity from syslog.h
   // because they collide with the names of our own severity levels. Therefore
-- 
2.30.2

