From cb7aa74643a42631d25644b83d084ca9c7c0e60a Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Mon, 27 Jul 2020 15:33:23 +0900
Subject: [PATCH 59/85] Apply adobe flash

---
 gooroom/browser/gooroom_pref.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index c4437b4d972c..8be66bb54e38 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -107,13 +107,14 @@ void GooroomPref::setup() {
     }
   }
   if (!use_user_data_dir) {
-    const char** restartoption = new const char*[argv.size() + 2];
-    restartoption[argv.size() + 1] = NULL;
+    const char** restartoption = new const char*[argv.size() + 3];
+    restartoption[argv.size() + 2] = NULL;
     restartoption[0] = "gooroom-browser";
     for (int j = 1; j < (int)argv.size(); j++) {
       restartoption[j] = argv[j].c_str();
     }
     restartoption[argv.size()] = trust_user_data_dir_option.c_str();
+    restartoption[argv.size() + 1] = flash_dir.c_str();
     execv(exe_path.c_str(), (char**)restartoption);
     exit(0);
   }
-- 
2.20.1

