From dc48fa535ea5b7aedf82c27ce1d9f8bc4b82c14a Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Mon, 27 Jul 2020 15:41:46 +0900
Subject: [PATCH 63/84] Set maximum length of the extension display name

---
 extensions/common/extension.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index ab0e3dcc8..83f3657fd 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -612,6 +612,7 @@ bool Extension::LoadRequiredFeatures(base::string16* error) {
 
 bool Extension::LoadName(base::string16* error) {
   base::string16 localized_name;
+  size_t display_max_len_ = 512;
   if (!manifest_->GetString(keys::kName, &localized_name)) {
     *error = base::ASCIIToUTF16(errors::kInvalidName);
     return false;
@@ -622,6 +623,8 @@ bool Extension::LoadName(base::string16* error) {
       base::CollapseWhitespace(localized_name, true);
   base::i18n::SanitizeUserSuppliedString(&sanitized_name);
   display_name_ = base::UTF16ToUTF8(sanitized_name);
+  if (display_name_.length() > display_max_len_)
+    display_name_.resize(display_max_len_);
   return true;
 }
 
-- 
2.20.1

