From 30a4174c4c60edfe6df740e1656d1942fd8ad6c2 Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Fri, 5 Jun 2020 15:11:38 +0900
Subject: [PATCH 67/79] [BTS:1877] option with url bug

---
 gooroom/browser/gooroom_pref.cc | 26 --------------------------
 1 file changed, 26 deletions(-)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index e962b01f0c52..c4437b4d972c 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -105,32 +105,6 @@ void GooroomPref::setup() {
 
       break;
     }
-    if (GURL(*it).is_valid() == true) {
-      bool trusted = pfilterImpl->isTrustUrl(*it);
-      base::GetGooroomBase()->setTrustMode(trusted);
-      int pid;
-      if (trusted) {
-        pid = fork();
-        if (pid != 0) {
-          exit(0);
-        } else {
-          execl(exe_path.c_str(), "gooroom-browser",
-                trust_user_data_dir_option.c_str(), flash_dir.c_str(), GURL(*it).spec().c_str(),
-                NULL);
-          exit(0);
-        }
-      } else {
-        pid = fork();
-        if (pid != 0) {
-          exit(0);
-        } else {
-          execl(exe_path.c_str(), "gooroom-browser",
-                untrust_user_data_dir_option.c_str(),flash_dir.c_str(), GURL(*it).spec().c_str(),
-                NULL);
-          exit(0);
-        }
-      }
-    }
   }
   if (!use_user_data_dir) {
     const char** restartoption = new const char*[argv.size() + 2];
-- 
2.20.1

