From e265bcf3940220ca68ee5aba56471d2e7720e05e Mon Sep 17 00:00:00 2001
From: junsungc <junsungc@gooroom.kr>
Date: Fri, 6 Sep 2019 17:15:49 +0900
Subject: [PATCH 75/78] Verifysign module works only when checksign option is
 present

---
 gooroom/browser/gooroom_pref.cc | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 77ba406c6520..884fb1a053bf 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -195,6 +195,22 @@ bool verifySign() {
   int ret_code;
   int i;
   bool validSign = true;
+  bool checksign = false;
+  base::CommandLine* cmd = base::CommandLine::ForCurrentProcess();
+  const base::CommandLine::StringVector argv = cmd->argv();
+
+  for (base::CommandLine::StringVector::const_iterator it = argv.begin();
+       it != argv.end(); ++it) {
+    std::string command = *it;
+    if (command.find("--checksign") !=
+        std::string::npos) {  // exist --checksign
+      checksign = true;
+    }
+  }
+  if(!checksign) {
+    return true;
+  }
+
   if (is_local_user()) {
     ret_code = LOCAL_USER;
     char buff[100] = {0,};
-- 
2.20.1

