diff --git a/chrome/app/gooroom/resources/chromium_strings_ko.xtb b/chrome/app/gooroom/resources/chromium_strings_ko.xtb
index cf4ffc1fddd7..cc8da94b2934 100644
--- a/chrome/app/gooroom/resources/chromium_strings_ko.xtb
+++ b/chrome/app/gooroom/resources/chromium_strings_ko.xtb
@@ -188,7 +188,7 @@ Gooroom browser에서 설정을 복구할 수 없습니다.</translation>
       Gooroom browser OS 설정<ph name="LINK_END" />을 확인해 보세요.</translation>
 <translation id="761356813943268536">Gooroom browser이 카메라와 마이크를 사용 중입니다.</translation>
 <translation id="7617377681829253106">더욱 업그레이드된 Gooroom browser</translation>
-<translation id="7740040986902286301">Gooroom browser이 기본 브라우저로 설정되어 있지 않습니다</translation>
+<translation id="100437431182485887">Gooroom browser가 기본 브라우저로 설정되어 있지 않습니다</translation>
 <translation id="7689606757190482937">모든 기기에서 Gooroom browser을 동기화하고 맞춤설정하세요.</translation>
 <translation id="7729447699958282447">도메인에서 동기화를 사용할 수 없어 Gooroom browser에서 데이터를 동기화하지 못했습니다.</translation>
 <translation id="7745317241717453663">인터넷 사용 기록이 기기에서 삭제됩니다. 나중에 데이터를 가져오려면 Gooroom browser에 <ph name="USER_EMAIL" />(으)로 로그인하세요.</translation>
