diff --git a/chrome/browser/ui/views/bookmarks/bookmark_bubble_view.cc b/chrome/browser/ui/views/bookmarks/bookmark_bubble_view.cc
index 36bd9c4e2538..9a2f4cef01a7 100644
--- a/chrome/browser/ui/views/bookmarks/bookmark_bubble_view.cc
+++ b/chrome/browser/ui/views/bookmarks/bookmark_bubble_view.cc
@@ -305,6 +305,8 @@ std::unique_ptr<views::View> BookmarkBubbleView::CreateSigninPromoView() {
 #if defined(OS_CHROMEOS)
   // ChromeOS does not show the signin promo.
   return nullptr;
+#elif defined(GOOROOM_BUILD)
+  return nullptr;
 #else
   if (!SyncPromoUI::ShouldShowSyncPromo(profile_))
     return nullptr;
