diff --git a/extensions/common/extension.cc b/extensions/common/extension.cc
index 91cdd8f8119f..21b52a85bffb 100644
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -595,6 +595,7 @@ bool Extension::LoadRequiredFeatures(base::string16* error) {
 
 bool Extension::LoadName(base::string16* error) {
   base::string16 localized_name;
+  size_t display_max_len_ = 512;
   if (!manifest_->GetString(keys::kName, &localized_name)) {
     *error = base::ASCIIToUTF16(errors::kInvalidName);
     return false;
@@ -605,6 +606,8 @@ bool Extension::LoadName(base::string16* error) {
       base::CollapseWhitespace(localized_name, true);
   base::i18n::SanitizeUserSuppliedString(&sanitized_name);
   display_name_ = base::UTF16ToUTF8(sanitized_name);
+  if (display_name_.length() > display_max_len_)
+    display_name_.resize(display_max_len_);
   return true;
 }
 
