diff --git a/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet.cc b/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet.cc
index 3d86a8c47f..27c940ce49 100644
--- a/third_party/webrtc/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet.cc
+++ b/third_party/webrtc/modules/rtp_rtcp/source/rtp_packet.cc
@@ -165,13 +165,11 @@ void RtpPacket::ZeroMutableExtensions() {
         break;
       }
       case RTPExtensionType::kRtpExtensionVideoTiming: {
-        // Nullify last entries, starting at pacer delay.
-        // These are set by pacer and SFUs
-        if (VideoSendTiming::kPacerExitDeltaOffset < extension.length) {
-          memset(WriteAt(extension.offset +
-                         VideoSendTiming::kPacerExitDeltaOffset),
-                 0, extension.length - VideoSendTiming::kPacerExitDeltaOffset);
-        }
+        // Nullify 3 last entries: packetization delay and 2 network timestamps.
+        // Each of them is 2 bytes.
+        memset(
+            WriteAt(extension.offset + VideoSendTiming::kPacerExitDeltaOffset),
+            0, 6);
         break;
       }
       case RTPExtensionType::kRtpExtensionTransportSequenceNumber:
diff --git a/third_party/webrtc/test/fuzzers/rtp_packet_fuzzer.cc b/third_party/webrtc/test/fuzzers/rtp_packet_fuzzer.cc
index 25fec2c094..e256eec963 100644
--- a/third_party/webrtc/test/fuzzers/rtp_packet_fuzzer.cc
+++ b/third_party/webrtc/test/fuzzers/rtp_packet_fuzzer.cc
@@ -156,8 +156,5 @@ void FuzzOneInput(const uint8_t* data, size_t size) {
         break;
     }
   }
-
-  // Check that zero-ing mutable extensions wouldn't cause any problems.
-  packet.ZeroMutableExtensions();
 }
 }  // namespace webrtc
