diff --git a/components/history/core/browser/top_sites_impl.cc b/components/history/core/browser/top_sites_impl.cc
index 23d1284d12a1..5f12bfac76b7 100644
--- a/components/history/core/browser/top_sites_impl.cc
+++ b/components/history/core/browser/top_sites_impl.cc
@@ -430,7 +430,12 @@ void TopSitesImpl::OnGotMostVisitedURLs(MostVisitedURLList sites) {
 
   // Set |top_sites_| directly so that SetTopSites() diffs correctly.
   top_sites_ = sites;
+#if defined(GOOROOM_BUILD)
+  MostVisitedURLList fakeSites;
+  SetTopSites(std::move(fakeSites), CALL_LOCATION_FROM_ON_GOT_MOST_VISITED_URLS);
+#else
   SetTopSites(std::move(sites), CALL_LOCATION_FROM_ON_GOT_MOST_VISITED_URLS);
+#endif
 
   MoveStateToLoaded();
 
