diff --git a/DEPS b/DEPS
index 75701becdbf7..51d24d43e332 100644
--- a/DEPS
+++ b/DEPS
@@ -1503,7 +1503,7 @@ deps = {
     Var('chromium_git') + '/external/khronosgroup/webgl.git' + '@' + 'dd55f3ca8f2ea716ca917a4aaf36f0729fe902b1',
 
   'src/third_party/webrtc':
-    Var('webrtc_git') + '/src.git' + '@' + 'ddc7578b751b05b5078c41f3973aa8fafdd543b7',
+    Var('webrtc_git') + '/src.git' + '@' + 'refs/branch-heads/3987',
 
   'src/third_party/libgifcodec':
      Var('skia_git') + '/libgifcodec' + '@'+  Var('libgifcodec_revision'),
