diff --git a/chrome/browser/search/ntp_features.cc b/chrome/browser/search/ntp_features.cc
index 099ffc627c44..a51e70a7a296 100644
--- a/chrome/browser/search/ntp_features.cc
+++ b/chrome/browser/search/ntp_features.cc
@@ -28,7 +28,7 @@ const base::Feature kNtpCustomizationMenuV2{"NtpCustomizationMenuV2",
 // directly (i.e. not be a "fake" box) and search results will show directly
 // below the non-fake input ("realbox").
 const base::Feature kNtpRealbox{"NtpRealbox",
-                                base::FEATURE_DISABLED_BY_DEFAULT};
+                                base::FEATURE_ENABLED_BY_DEFAULT};
 
 bool IsNtpRealboxEnabled() {
   return base::FeatureList::IsEnabled(kNtpRealbox) ||
