diff --git a/chrome/app/gooroom/resources/generated_resources_ko.xtb b/chrome/app/gooroom/resources/generated_resources_ko.xtb
index a2ac73861fd4..f0c63a5b0bc8 100644
--- a/chrome/app/gooroom/resources/generated_resources_ko.xtb
+++ b/chrome/app/gooroom/resources/generated_resources_ko.xtb
@@ -2835,7 +2835,7 @@
 <translation id="473775607612524610">업데이트</translation>
 <translation id="4739639199548674512">티켓</translation>
 <translation id="4743260470722568160"><ph name="BEGIN_LINK" />애플리케이션 업데이트 방법 알아보기<ph name="END_LINK" /></translation>
-<translation id="4744981231093950366">{NUM_TABS,plural, =1{사이트 숨기기 취소}other{사이트 숨기기 취소}}</translation>
+<translation id="4744981231093950366">{NUM_TABS,plural, =1{소리 켜기}other{소리 켜기}}</translation>
 <translation id="4746351372139058112">메시지</translation>
 <translation id="4750394297954878236">추천</translation>
 <translation id="475088594373173692">첫 번째 사용자</translation>
@@ -2994,7 +2994,7 @@
 <translation id="4972129977812092092">프린터 수정</translation>
 <translation id="4972164225939028131">잘못된 비밀번호</translation>
 <translation id="4973307593867026061">프린터 추가</translation>
-<translation id="4973325300212422370">{NUM_TABS,plural, =1{사이트 숨기기}other{사이트 숨기기}}</translation>
+<translation id="4973325300212422370">{NUM_TABS,plural, =1{소리 끄기}other{소리 끄기}}</translation>
 <translation id="4974733135013075877">종료 및 자녀 보호 잠금</translation>
 <translation id="4976009197147810135">수직 분할</translation>
 <translation id="4977942889532008999">액세스 확인</translation>
