diff --git a/chrome/app/gooroom/resources/generated_resources_ko.xtb b/chrome/app/gooroom/resources/generated_resources_ko.xtb
index 3b0b781..4bce67e 100644
--- a/chrome/app/gooroom/resources/generated_resources_ko.xtb
+++ b/chrome/app/gooroom/resources/generated_resources_ko.xtb
@@ -2272,6 +2272,7 @@
 <translation id="4316850752623536204">개발자 웹사이트</translation>
 <translation id="4320177379694898372">인터넷에 연결되지 않음</translation>
 <translation id="4321442524549817090">URL 표시(&amp;H)</translation>
+<translation id="4322394346347055525">다른 탭 닫기</translation>
 <translation id="4324577459193912240">파일 다운로드가 완료되지 않음</translation>
 <translation id="4325237902968425115"><ph name="LINUX_APP_NAME" /> 제거 중...</translation>
 <translation id="4330191372652740264">얼음물</translation>
