diff --git a/chrome/browser/ui/ui_features.cc b/chrome/browser/ui/ui_features.cc
index 5924106333bb..11ead2d0b2b2 100644
--- a/chrome/browser/ui/ui_features.cc
+++ b/chrome/browser/ui/ui_features.cc
@@ -75,7 +75,7 @@ const base::Feature kTabGroupsCollapseFreezing{
 // Enables the feedback through the tab group editor bubble.
 // https://crbug.com/1067062
 const base::Feature kTabGroupsFeedback{"TabGroupsFeedback",
-                                       base::FEATURE_ENABLED_BY_DEFAULT};
+                                       base::FEATURE_DISABLED_BY_DEFAULT};
 
 // Enables popup cards containing tab information when hovering over a tab.
 // https://crbug.com/910739
