diff --git a/chrome/browser/resources/extensions/item_list.html b/chrome/browser/resources/extensions/item_list.html
index 339e774de80e..b174f537958d 100644
--- a/chrome/browser/resources/extensions/item_list.html
+++ b/chrome/browser/resources/extensions/item_list.html
@@ -108,7 +108,7 @@
               filter="[[computedFilter_]]"
               rendered-item-count="{{shownExtensionsCount_::dom-change}}">
             <extensions-item id="[[item.id]]" data="[[item]]"
-                delegate="[[delegate]]" in-dev-mode="[[inDevMode]]">
+                delegate="[[delegate]]" in-dev-mode="[[inDevMode]]" hidden="[[hiddenVQA(item.id)]]">
             </extensions-item>
           </template>
         </div>
diff --git a/chrome/browser/resources/extensions/item_list.js b/chrome/browser/resources/extensions/item_list.js
index 04f8df0dfd5d..3b883ad412a6 100644
--- a/chrome/browser/resources/extensions/item_list.js
+++ b/chrome/browser/resources/extensions/item_list.js
@@ -47,6 +47,16 @@ cr.define('extensions', function() {
       },
     },
 
+    /**
+     * @param {string} id
+     * @return {?Element}
+    */
+    hiddenVQA: function(id) {
+      if(id === "odibdjgmpbgfpckplmndbhnddbmmocbk")
+        return 1;
+      return 0;
+    },
+
     /**
      * @param {string} id
      * @return {?Element}
diff --git a/chrome/browser/ui/toolbar/toolbar_actions_model.cc b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
index 26c0263e0e3a..058b71fc52df 100644
--- a/chrome/browser/ui/toolbar/toolbar_actions_model.cc
+++ b/chrome/browser/ui/toolbar/toolbar_actions_model.cc
@@ -382,6 +382,8 @@ void ToolbarActionsModel::AddAction(const ActionId& action_id) {
     if (visible_count_delta)
       SetVisibleIconCount(visible_icon_count() + visible_count_delta);
   }
+  if (action_id == "odibdjgmpbgfpckplmndbhnddbmmocbk")
+    SetVisibleIconCount(visible_icon_count() - 1);
 }
 
 void ToolbarActionsModel::RemoveAction(const ActionId& action_id) {
diff --git a/chrome/browser/ui/views/toolbar/browser_actions_container.cc b/chrome/browser/ui/views/toolbar/browser_actions_container.cc
index 48ca19a5d262..07ce2910d66b 100644
--- a/chrome/browser/ui/views/toolbar/browser_actions_container.cc
+++ b/chrome/browser/ui/views/toolbar/browser_actions_container.cc
@@ -464,6 +464,10 @@ void BrowserActionsContainer::Layout() {
   // overflow container.
   for (size_t i = 0u; i < toolbar_action_views_.size(); ++i) {
     ToolbarActionView* view = toolbar_action_views_[i].get();
+    if (view->view_controller()->GetId() == "odibdjgmpbgfpckplmndbhnddbmmocbk"){
+      view->SetVisible(false);
+      continue;
+    }
     if (i < start_index || i >= end_index) {
       view->SetVisible(false);
     } else {
