diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 3ea4f033f..5f7b7d1b8 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -393,7 +393,8 @@ void GooroomPref::setup() {
   if(base::CommandLine::ForCurrentProcess()->HasSwitch(switches::kPpapiFlashPath))
   {
     std::string flashVersion = FlashVersion::GetFlashVersion();
-    base::CommandLine::ForCurrentProcess()->AppendSwitchASCII(switches::kPpapiFlashVersion, flashVersion);
+    if(!flashVersion.empty())
+      base::CommandLine::ForCurrentProcess()->AppendSwitchASCII(switches::kPpapiFlashVersion, flashVersion);
   }
 
 }
diff --git a/gooroom/flash/flash_version.cc b/gooroom/flash/flash_version.cc
index eb352296e..f5e73e1e3 100644
--- a/gooroom/flash/flash_version.cc
+++ b/gooroom/flash/flash_version.cc
@@ -2,7 +2,7 @@
 
 std::string FlashVersion::GetFlashVersion()
 {
-  std::string line;
+  std::string line = "";
   std::ifstream flash_manifest_json("/usr/lib/adobe-flashplugin/manifest.json");
   if (flash_manifest_json.is_open())
   {
@@ -29,4 +29,5 @@ std::string FlashVersion::GetFlashVersion()
       }
     }
   }
+  return line;
 }
