diff --git a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
index 55793c62eff7..6a1982a7a667 100644
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -29,6 +29,7 @@ bool IsComponentExtensionAllowlisted(const std::string& extension_id) {
     extension_misc::kInAppPaymentsSupportAppId,
     extension_misc::kCastExtensionIdRelease,
     extension_misc::kPdfExtensionId,
+    extension_misc::kOdfExtensionId,
 #if defined(OS_CHROMEOS)
     extension_misc::kAssessmentAssistantExtensionId,
     extension_misc::kAccessibilityCommonExtensionId,
