diff --git a/chrome/browser/resources/local_ntp/local_ntp.js b/chrome/browser/resources/local_ntp/local_ntp.js
index abbdf650f67e..083bc89d793a 100644
--- a/chrome/browser/resources/local_ntp/local_ntp.js
+++ b/chrome/browser/resources/local_ntp/local_ntp.js
@@ -998,7 +998,6 @@ function handlePostMessage(event) {
         $(customBackgrounds.IDS.CUSTOM_LINKS_RESTORE_DEFAULT).tabIndex =
             (args.showRestoreDefault ? 0 : -1);
       }
-      $(IDS.OGB).classList.add(CLASSES.SHOW_ELEMENT);
     }
   } else if (cmd === 'tileBlacklisted') {
     if (configData.isGooglePage) {
@@ -1437,12 +1436,10 @@ function injectOneGoogleBar(ogb) {
   renderOneGoogleBarTheme();
 
   const ogElem = $('one-google');
-  ogElem.innerHTML = ogb.barHtml;
 
   const afterBarScript = document.createElement('script');
   afterBarScript.type = 'text/javascript';
   afterBarScript.appendChild(document.createTextNode(ogb.afterBarScript));
-  ogElem.parentNode.insertBefore(afterBarScript, ogElem.nextSibling);
 
   $('one-google-end-of-body').innerHTML = ogb.endOfBodyHtml;
 
