diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index 884fb1a..bf96508 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -1,7 +1,6 @@
 #include "gooroom_pref.h"
 
 #include <fstream>
-#include <gtk/gtk.h>
 
 #include "verify_signature.h"
 #include "base/files/file_enumerator.h"
@@ -37,39 +36,6 @@ bool is_file_exist(const char *fileName)
     return infile.good();
 }
 
-void noVerifySignAlarm() {
-    GtkWidget *window;
-    GtkWidget *label;
-
-    /* Create the main, top level window */
-    window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
-
-    /* Give it the title */
-    gtk_window_set_title(GTK_WINDOW(window), "Info Bars");
-    gtk_window_set_default_size(GTK_WINDOW(window), 230, 50);
-
-    /*
-    ** Map the destroy signal of the window to gtk_main_quit;
-    ** When the window is about to be destroyed, we get a notification and
-    ** stop the main GTK+ loop by returning 0
-    */
-    g_signal_connect(window, "destroy", G_CALLBACK(gtk_main_quit), NULL);
-
-    label = gtk_label_new("Gooroom-Browser : No sign verify. Check gooroom-agent service.");
-
-    /* Plot the label onto the main window */
-    gtk_container_add(GTK_CONTAINER(window), label);
-
-    /* Make sure that everything, window and label, are visible */
-    gtk_widget_show_all(window);
-
-    /*
-    ** Start the main loop, and do nothing (block) until
-    ** the application is closed
-    */
-    gtk_main();
-}
-
 int is_local_user() {
   FILE *fp = NULL;
   char input[1024];
@@ -380,7 +346,6 @@ void GooroomPref::setup() {
   // Gooroom-Browser policy Sign module
   bool validSign = verifySign();
   if (!validSign) {
-    noVerifySignAlarm();
     exit(0);
   }
 
