diff --git a/build/config/linux/gtk2/BUILD.gn b/build/config/linux/gtk2/BUILD.gn
index a58a4b6..4d696c2 100644
--- a/build/config/linux/gtk2/BUILD.gn
+++ b/build/config/linux/gtk2/BUILD.gn
@@ -28,6 +28,7 @@ group("gtk2") {
   visibility = [
     "//build/config/linux/gtk",
     "//chrome/browser/ui/libgtkui:*",
+    "//gooroom/browser:gooroom_browser",
   ]
   public_configs = [ ":gtk2_internal_config" ]
 }
