diff --git a/gooroom/browser/gooroom_pref.cc b/gooroom/browser/gooroom_pref.cc
index e67522e..1c76045 100644
--- a/gooroom/browser/gooroom_pref.cc
+++ b/gooroom/browser/gooroom_pref.cc
@@ -2,6 +2,7 @@

 #include <fstream>
 #include <gtk/gtk.h>
+#include <sys/stat.h>

 #include "verify_signature.h"
 #include "gooroom/base/gooroom_base.h"
diff --git a/gooroom/browser/verify_signature.cc b/gooroom/browser/verify_signature.cc
index 0c0354f..d6f5297 100644
--- a/gooroom/browser/verify_signature.cc
+++ b/gooroom/browser/verify_signature.cc
@@ -258,7 +258,7 @@ do_task(const char *task)
 	Py_DECREF(args);
 	if (!value) goto ERR_CALL;
 
-	char_value = PyUnicode_AsUTF8AndSize(value, &value_size);
+	char_value = (char *)PyUnicode_AsUTF8AndSize(value, &value_size);
 	return_value = (char *)calloc(1, value_size+1);
 	memcpy(return_value, char_value, value_size);
 
