description: AV1 is not supported by libva in buster
author: Michael Gilbert <mgilbert@debian.org>

--- a/media/gpu/vaapi/vaapi_unittest.cc
+++ b/media/gpu/vaapi/vaapi_unittest.cc
@@ -43,7 +43,9 @@ base::Optional<VAProfile> ConvertToVAPro
     {VP8PROFILE_ANY, VAProfileVP8Version0_3},
     {VP9PROFILE_PROFILE0, VAProfileVP9Profile0},
     {VP9PROFILE_PROFILE2, VAProfileVP9Profile2},
+#if BUILDFLAG(USE_VAAPI)
     {AV1PROFILE_PROFILE_MAIN, VAProfileAV1Profile0},
+#endif
 #if BUILDFLAG(ENABLE_PLATFORM_HEVC)
     {HEVCPROFILE_MAIN, VAProfileHEVCMain},
     {HEVCPROFILE_MAIN10, VAProfileHEVCMain10},
@@ -68,7 +70,9 @@ base::Optional<VAProfile> StringToVAProf
     {"VAProfileVP8Version0_3", VAProfileVP8Version0_3},
     {"VAProfileVP9Profile0", VAProfileVP9Profile0},
     {"VAProfileVP9Profile2", VAProfileVP9Profile2},
+#if BUILDFLAG(USE_VAAPI)
     {"VAProfileAV1Profile0", VAProfileAV1Profile0},
+#endif
 #if BUILDFLAG(ENABLE_PLATFORM_HEVC)
     {"VAProfileHEVCMain", VAProfileHEVCMain},
     {"VAProfileHEVCMain10", VAProfileHEVCMain10},
--- a/media/gpu/vaapi/vaapi_wrapper.cc
+++ b/media/gpu/vaapi/vaapi_wrapper.cc
@@ -402,7 +402,9 @@ const ProfileCodecMap& GetProfileCodecMa
           {VP9PROFILE_PROFILE2, VAProfileVP9Profile2},
       // VaapiWrapper does not support Profile 3.
       //{VP9PROFILE_PROFILE3, VAProfileVP9Profile3},
+#if BUILDFLAG(USE_VAAPI)
           {AV1PROFILE_PROFILE_MAIN, VAProfileAV1Profile0},
+#endif
         // VaapiWrapper does not support AV1 Profile 1.
         // {AV1PROFILE_PROFILE_HIGH, VAProfileAV1Profile1},
 #if BUILDFLAG(ENABLE_PLATFORM_HEVC)
@@ -439,8 +441,11 @@ bool IsVAProfileSupported(VAProfile va_p
 
 bool IsBlockedDriver(VaapiWrapper::CodecMode mode, VAProfile va_profile) {
   if (!IsModeEncoding(mode)) {
+#if BUILDFLAG(USE_VAAPI)
     return va_profile == VAProfileAV1Profile0 &&
            !base::FeatureList::IsEnabled(kVaapiAV1Decoder);
+#endif
+    return false;
   }
 
   // TODO(posciak): Remove once VP8 encoding is to be enabled by default.
